skimage/transform/histogram_matching.py

```diff
@@ -19,12 +19,17 @@ def _match_cumulative_cdf(source, template):
     return interp_a_values[src_unique_indices].reshape(source.shape)
 
 
-
 def match_histograms(image, reference, multichannel=False):
     """Adjust an image so that its cumulative histogram matches that of another.
 
     The adjustment is applied separately for each channel.
 
+    For color image assume to use another colour space then RGB since the colour
+    transitions are not very smooth. For example having red source image
+    and green template image will not produce expected result. Therefor convert
+    your images to "continuous" color space like HSV first and after histogram
+    matching back to RGB. [2]
```

Comment:
This could use some rewording:

> Note that for images in RGB space, the colour transitions are not very smooth. For example, having a red source image and a green template image may not produce the expected results as the per-channel histograms are unrelated. You may have better results by first converting your images to a continuous color space such as HSV prior to calling the function match_instograms [2]_.

Basically, I wouldn't make any hard statements about the performance of the algorithm for a specific use case.

Reply:
that why I did not d any code change just add a warning that the standards RGB is not the best colour space for histogram matching per channel

Reply:
Btw, what reworking do you have in mind? I am also fine to remove it completely...

Reply:
`assume to use another colour space` isn't very clear. I think you mean `RGB is not the best color space`.

`will not` -> `may not`

`Therefor` -> `Therefor`

`convert` -> `you may get better results by first` <----- key addition here is the word "may".

Reply:
`Therefor` -> `Therefor` ?

Reply:
gotta add that e ;)